tests/Composer/Test/Util/FilesystemTest.php

```diff
@@ -399,4 +399,32 @@ public function testCopyThenRemove(): void
         self::assertDirectoryDoesNotExist($this->workingDir . '/foo/bar', 'Still a directory: ' . $this->workingDir . '/foo/bar');
         self::assertDirectoryDoesNotExist($this->workingDir . '/foo', 'Still a directory: ' . $this->workingDir . '/foo');
     }
+
+    /**
+     * A Unix path is absolute if it starts with a forward-slash.
+     * A Windows path is absolute if it starts with a drive letter followed by a colon and a back-slash.
+     * A network path is always absolute and starts with two backslashes.
+     * A stream path is always absolute and starts with a scheme followed by "://".
+     */
+    public static function isAbsolutePathDataProvider(): array
+    {
+        return [
+            'unixPath' => ['/foo/bar', true],
+            'smbPath' => ['\\\\smb\\folder\\file.txt', true],
+            'windowsPath' => ['C:\\foo\\bar', true],
+            'streamPath' => ['file://path/to/whatever', true],
```

Comment:
I don't think this is true, file:// is not absolute per https://www.php.net/manual/en/wrappers.file.php

Reply:
```suggestion
            'streamPath' => ['custom://path/to/whatever', true],
```
Hmmm. I guess `file://` can be relative because the system has the context of the current process. I see on [bugs.php.net](https://bugs.php.net/bug.php?id=37555) that `chdir()` doesn't accept stream wrappers, so can we correctly say that all custom stream wrappers are absolute paths? `Filesystem::isAbsolutePath()` could be changed to strip `file://` before assessing those paths.

Reply:
Yup I'd think that is true for custom ones

Reply:
Addressed in ed968c0bdc94d10a24c9537f9de8de5eb72db6a2